core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/TinyLfuBlockCache.java

```diff
@@ -68,7 +73,10 @@ public long getMaxSize() {
 
   @Override
   public CacheEntry getBlock(String blockName) {
-    return cache.getIfPresent(blockName);
+    CacheEntry ce = cache.getIfPresent(blockName);
+    if (ce != null)
+      return ce;
+    return weakEvictionMap.get(blockName);
```

Comment:
You might prefer to promote the entry using a CacheLoader. Otherwise your recovery of a premature eviction would still allow the entry to be collected eventually.